static-analysis/teammates-mackerTest.xml

```diff
@@ -0,0 +1,159 @@
+                </allow>
+                
+                <!--Violations that need fixing-->
+                <allow>
+                    <from class="teammates.test.cases.logic.StudentsLogicTest" />
+                    <to class="teammates.storage.entity.CourseStudent" />
+                </allow>
+                <allow>
+                    <from class="teammates.test.cases.storage.FeedbackResponseCommentsDbTest" />
+                    <to class="teammates.storage.entity.FeedbackResponseComment" />
+                </allow>
+            </deny>
+        </access-rule>
+    </ruleset>
+    <ruleset name="TestDriverComponent: Only certain test cases can access GaeSimulation/BackDoorLogic">
+        <access-rule>
+            <deny>
+                <to>
+                    <include class="teammates.test.driver.GaeSimulation" />
+                    <include class="teammates.logic.backdoor.BackDoorLogic" />
+                </to>
+                <allow>
+                    <from>
+                        <include class="teammates.test.cases.storage.**" />
+                        <include class="teammates.test.cases.logic.**" />
+                        <include class="teammates.test.cases.action.**" />
+                        <include class="teammates.test.cases.automated.**" />
+                        <include class="teammates.test.cases.search.**" />
+                        <include class="teammates.test.cases.BaseComponentTestCase" />
+                        
+                        <!--Violation that needs fixing-->
+                        <include class="teammates.test.cases.pagedata.StudentFeedbackResultsPageDataTest" />
+                    </from>
+                </allow>
+                <allow>
+                    <!-- to import sample data for new instructor -->
+                    <from class="teammates.ui.controller.AdminInstructorAccountAddAction" />
+                    <to class="teammates.logic.backdoor.BackDoorLogic" />
+                </allow>
+                <allow>
+                    <from class="teammates.logic.backdoor.BackDoorServlet" />
+                    <to class="teammates.logic.backdoor.BackDoorLogic" />
+                </allow>
+            </deny>
+        </access-rule>
+    </ruleset>
+    <ruleset name="TestDriverComponent: BackDoor test driver is only for browser test">
+        <access-rule>
+            <deny>
+                <to class="teammates.test.driver.BackDoor" />
+                <allow>
+                    <from>
+                        <include class="teammates.test.cases.browsertests.**" />
+                        <include class="teammates.test.driver.BackDoor**$**" />
+                        <include class="teammates.test.cases.testdriver.BackDoorTest" />
+                        
+                        <!--Violation that needs fixing-->
+                        <include class="teammates.test.cases.BaseTestCaseWithDatastoreAccess" />
```

Comment:
The `BaseComponentTestCase` actually override methods in `BaseTestCaseWithDatastoreAccess` so that it can make use of `backDoorLogic` rather than `BackDoor`. I would say `BaseComponentTestCase` just should not extend from `BaseTestCaseWithDatastoreAccess` as it already has `BackDoorLogic`